test/features/respondToFieldsetsChange.test.ts

```diff
@@ -11,27 +11,75 @@ const SUPPORTING_COMPONENTS = [
   'FormModal',
 ];
 
+function getProps (componentName: string): any {
+  const { propsFactory } = COMPONENT_GENERATORS[componentName]
+    , onSave = jest.fn();
+
+  return omit(propsFactory.build({
+    onSave,
+  }), ['isVisible']);
+}
+
+async function changeProps (tester: Tester, props: object) {
+  tester.wrapper.setProps({ children: React.cloneElement(tester.wrapper.props().children, props)});
+  await tester.refresh();
+}
+
 // Tests that components respond to changes in fieldsets.
 describe('respondToFieldsetsChange', () => {
   SUPPORTING_COMPONENTS.forEach(componentName => {
-    it(`${componentName} successfully updates when fieldsets change`, async () => {
-      const { ComponentClass, propsFactory } = COMPONENT_GENERATORS[componentName]
-        , name = faker.lorem.sentence()
-        , props = propsFactory.build({
-            fieldSets: [[{ field: 'originalName', type: 'string' }]],
-            model: { name },
-        })
-        , newProps = propsFactory.build({
-          fieldSets: [[{ field: 'updatedName', type: 'string' }]],
-          model: { name },
-        });
-
-      const tester = await new Tester(ComponentClass, { props }).mount({async: true});
+    it(`${componentName} successfully updates when field name changes`, async () => {
+      const { ComponentClass } = COMPONENT_GENERATORS[componentName]
+        , props = getProps(componentName)
+        , oldProps = {...props, fieldSets: [[{ field: 'originalName', type: 'string' }]]}
+        , newProps = {...props, fieldSets: [[{ field: 'updatedName', type: 'string' }]]}
+        ;
+
+      const tester = await new Tester(ComponentClass, { props: oldProps }).mount({async: true});
 
       expect(tester.find('input[id="originalName"]').length).toBe(1);
-      tester.wrapper.setProps({ children: React.cloneElement(tester.wrapper.props().children, newProps)});
-      await tester.refresh();
+      expect(tester.find('input[id="updatedName"]').length).toBe(0);
+
+      tester.submit();
+      expect(props.onSave).toHaveBeenCalledWith({ originalName: '' });
+
+      props.onSave.mockClear();
+
+      await changeProps(tester, newProps);
+
+      expect(tester.find('input[id="originalName"]').length).toBe(0);
       expect(tester.find('input[id="updatedName"]').length).toBe(1);
+
+      tester.submit();
+      expect(props.onSave).toHaveBeenCalledWith({ updatedName: '' });
+    });
+
+    it(`${componentName} successfully updates when number of fieldsets change`, async () => {
```

Comment:
Since field sets is camel-cased throughout the app, I would expect it to be `field sets`, but I recognize that that seems counterintuitive

Reply:
I'll make it FieldSets, since that's the name of the interface for this type of object.